### PR TITLE
Checksum validate disk all endpoints

### PR DESCRIPTION
**Pull request by SaravShah**
- Made it so methods took `endpoint_name` instead of `storage_dir` to be more synonymous to c2m and m2c checks. 
- removed `algorithm` as a parameter since it is not used.
- Updated README with the given changes

connects #578

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -62,6 +62,7 @@ Metrics/AbcSize:
     - 'app/services/preserved_object_handler.rb' # methods still readable; shameless green
     - 'app/services/audit_results.rb' # method still readable
     - 'lib/audit/catalog_to_moab.rb' # .check_version_all_dirs is decidedly readable
+    - 'lib/audit/checksum.rb' # # .validate_disk_all)dirs is decidedly readable
```
Comment (ndushay) [inline .rubocop.yml]: double # # # # #  and some other typo, too.

File: .rubocop.yml [hunk 2]
```diff
@@ -130,6 +131,7 @@ Naming/FileName:
 Performance/HashEachMethods:
   Exclude:
     - lib/audit/catalog_to_moab.rb # it's a tiny config object; code is clearer as is
+    - lib/audit/checksum.rb #it's a tiny config object; code is clearer as is
```
Comment (ndushay) [inline .rubocop.yml]: missing a space after # but I don't really care.  I'm just a little OCD.

File: lib/audit/checksum.rb [hunk 3]
```diff
@@ -8,22 +8,31 @@ class Checksum
   # rubocop:disable Style/EmptyMethod:
   # TODO: implement this;  we begin with a placeholder
 
-  def self.validate_disk(last_checked_b4, endpoint, algorithm="MD5")
+  def self.validate_disk(last_checked_b4, storage_dir)
```
Comment (ndushay) [inline lib/audit/checksum.rb]: so we have a ticket to wire this method up to the actual code when it's ready, yes??  We could put in the timestamp code and so on analogous to M2C / C2M checks now??  (or put that info in ticket?)

File: spec/lib/audit/checksum_spec.rb [hunk 4]
```diff
@@ -28,30 +29,37 @@
...
-      described_class.validate_disk_all_endpoints('2018-02-05 21:37:23 UTC', 'MD5')
-      skip 'we should figure out what they are and test them'
+      Settings.moab.storage_roots.each do |storage_root|
+        expect(described_class).to receive(:validate_disk).with(
+          last_checked_b4_date,
+          "#{storage_root[1]}/#{Settings.moab.storage_trunk}"
+        )
+      end
+      subject
     end
```
Comment (ndushay) [inline spec/lib/audit/checksum_spec.rb]: if we wanted to go nuts we could add tests looking for the start and end statements in the logs.  But I doubt it's worth it.

File: .rubocop.yml (hunk 2 above)
Comment (jmartin-sul) [reply .rubocop.yml]: ha, this also triggered my OCD, despite the fact that lack of space used to be my default comment style.  rubocop has assimilated me.

File: lib/audit/checksum.rb [hunk 5]
```diff
@@ -8,22 +8,22 @@ class Checksum
...
```
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply lib/audit/checksum.rb]: and indeed, it is part of the query in this PR: https://github.com/sul-dlss/preservation_catalog/pull/534

File: Rakefile [hunk 6]
```diff
@@ -165,39 +165,37 @@ task :c2m_check_version_all_dirs, [:last_checked_b4_date, :profile] => [:environ
   $stdout.flush
 end
 
-desc "Fire off checksum validation on a single endpoint"
-task :cv_single_endpoint, [:last_checked_b4, :endpoint, :algorithm, :profile] => [:environment] do |_t, args|
+desc "Fire off checksum validation on a single storage root"
+task :cv_single_dir, [:last_checked_b4, :storage_dir, :profile] => [:environment] do |_t, args|
```
[the remark this answers is not in the document]
Comment (SaravShah) [reply Rakefile]: ah yeah - i got confused and  had it as an endpoint name at first but for some reason i changed it to a storage_dir i have no idea why... i'll change it back :)

File: lib/audit/checksum.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (SaravShah) [reply lib/audit/checksum.rb]: ah - so i can just remove the last_checked_b4 paramter from both ` validate_disk_all_dirs` and `validate_disk` in the check ChecksumValidator class too?

[the remark this answers is not in the document]
Comment (SaravShah) [reply lib/audit/checksum.rb]: also i can do this fix in my other ticket that has to do with manifestInv stuff

File: lib/audit/checksum.rb (hunk 3 above)
Comment (SaravShah) [reply lib/audit/checksum.rb]: i'll go ahead a ticket it, #595

File: Rakefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (SaravShah) [reply Rakefile]: i did this in my other PR too lol #591

File: lib/audit/checksum.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (SaravShah) [reply lib/audit/checksum.rb]: talked via Slack

File: Rakefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (SaravShah) [reply Rakefile]: talked via Slack! ✔️

**Comment by ndushay** [comment]
I'm fine with this being closed/merged, but I'm left wondering:  will we ever want to run fixity based on a date param, rather than by fixity_ttl on the preservation policy?

I guess we can cross that bridge when we come to it ...

Let's get some performance data!   That should help us, too.

**Comment by jmartin-sul** [comment]
yeah, i was wondering the same, and also figured we could cross that bridge later.  as it is, it seems like this PR pushes things forward with no downsides.

**Pull request merged**